pandas/io/excel.py

```diff
@@ -66,27 +69,164 @@ def get_writer(engine_name):
         raise ValueError("No Excel writer '%s'" % engine_name)
 
 
+class BaseFile(object):
+    """Base class for excel readers
+
+    A file class can be initialized even if the engine is not installed.
+    If the engine is not installed, io_class and _open_workbook are
+    type(None) and None respectively. When attempting to use open_workbook
+    while _open_workbook is None, the relevant ImportError is raised. When
+    calling is_type or is_ext, False is retunred when the reader is not
+    installed.
+    """
+    # added in order to deal with both Open Document Format and MS Excel
+    # fil types, see PR #9070
+
+    def __init__(self, engine, extensions):
+        """Set the engine name, and extension. If the engine is not installed,
+        io_class and _open_workbook are type(None) and None respectively,
+        but the ImportError is ingored at the init stage.
+        """
+        self.engine = engine
+        self.extensions = extensions
+        self.io_class = type(None)
+        self._open_workbook = None
+        # fail silently if engine is not presented on init, only raise when
+        # attempting to use it when opening a workbook
+        try:
+            self.load_engine()
```

Comment:
But maybe this approach is not the most lean, clean and readable (based on your comments at least ;-). What I see as the challenge is that a user could try to read a spreadsheet that is supported by pandas, but not installed. I could try to re-factor this in another way. Instead of having some kind of fall back logic in the base/sub class of the reader, there could be something that registers which reader dependencies are a installed. When knowing which readers are available and which ones are not, I assume this could be made nicer. Do you think it is necessary to revise the current strategy?